keyboards/qwertykeys/qk100/ansi/ansi.c

```diff
@@ -0,0 +1,197 @@
+    {0, CS32_SW9, CS33_SW9, CS31_SW9}, /* RGB50-7 */
+    {0, CS35_SW9, CS36_SW9, CS34_SW9}, /* RGB51-8 */
+    {0, CS38_SW9, CS39_SW9, CS37_SW9}, /* RGB52-9 */
+    {0, CS2_SW4, CS3_SW4, CS1_SW4},    /* RGB53-CAPS---ROW3*/
+    {0, CS5_SW4, CS6_SW4, CS4_SW4},    /* RGB54-A-- */
+    {0, CS8_SW4, CS9_SW4, CS7_SW4},    /* RGB55-S-- */
+    {0, CS11_SW4, CS12_SW4, CS10_SW4}, /* RGB56-D */
+    {0, CS14_SW4, CS15_SW4, CS13_SW4}, /* RGB57-F */
+    {0, CS17_SW4, CS18_SW4, CS16_SW4}, /* RGB58-G */
+    {0, CS20_SW4, CS21_SW4, CS19_SW4}, /* RGB59-H */
+    {0, CS23_SW4, CS24_SW4, CS22_SW4}, /* RGB60-J */
+    {0, CS26_SW4, CS27_SW4, CS25_SW4}, /* RGB61-K */
+    {0, CS29_SW4, CS30_SW4, CS28_SW4}, /* RGB62-L */
+    {0, CS32_SW4, CS33_SW4, CS31_SW4}, /* RGB63-;: */
+    {0, CS35_SW4, CS36_SW4, CS34_SW4}, /* RGB64-'" */
+    {0, CS38_SW4, CS39_SW4, CS37_SW4}, /* RGB65-ENTER */
+    {0, CS11_SW7, CS12_SW7, CS10_SW7}, /* RGB66-Page Down */
+    {0, CS14_SW7, CS15_SW7, CS13_SW7}, /* RGB67-4 */
+    {0, CS17_SW7, CS18_SW7, CS16_SW7}, /* RGB68-5 */
+    {0, CS20_SW7, CS21_SW7, CS19_SW7}, /* RGB69-6 */
+    {0, CS23_SW7, CS24_SW7, CS22_SW7}, /* RGB70-+ */
+    {0, CS2_SW5, CS3_SW5, CS1_SW5},    /* RGB71-LSF --ROW4*/
+    {0, CS5_SW5, CS6_SW5, CS4_SW5},    /* RGB72-Z */
+    {0, CS8_SW5, CS9_SW5, CS7_SW5},    /* RGB73-X */
+    {0, CS11_SW5, CS12_SW5, CS10_SW5}, /* RGB74-C */
+    {0, CS14_SW5, CS15_SW5, CS13_SW5}, /* RGB75-V */
+    {0, CS17_SW5, CS18_SW5, CS16_SW5}, /* RGB76-B */
+    {0, CS20_SW5, CS21_SW5, CS19_SW5}, /* RGB77-N */
+    {0, CS23_SW5, CS24_SW5, CS22_SW5}, /* RGB78-M */
+    {0, CS26_SW5, CS27_SW5, CS25_SW5}, /* RGB79-,< */
+    {0, CS29_SW5, CS30_SW5, CS28_SW5}, /* RGB80->. */
+    {0, CS32_SW5, CS33_SW5, CS31_SW5}, /* RGB81-?/ */
+    {0, CS35_SW5, CS36_SW5, CS34_SW5}, /* RGB82-RSF */
+    {0, CS38_SW5, CS39_SW5, CS37_SW5}, /* RGB83-UP */
+    {0, CS14_SW8, CS15_SW8, CS13_SW8}, /* RGB84-1*/
+    {0, CS17_SW8, CS18_SW8, CS16_SW8}, /* RGB85-2 */
+    {0, CS20_SW8, CS21_SW8, CS19_SW8}, /* RGB86-3 */
+    {0, CS23_SW8, CS24_SW8, CS22_SW8}, /* RGB87-Enter */
+    {0, CS2_SW6, CS3_SW6, CS1_SW6},    /* RGB88-lct-- ROW5*/
+    {0, CS5_SW6, CS6_SW6, CS4_SW6},    /* RGB89-lwin */
+    {0, CS8_SW6, CS9_SW6, CS7_SW6},    /* RGB90-lalt */
+    {0, CS11_SW6, CS12_SW6, CS10_SW6}, /* RGB91-sp0 */
+    {0, CS17_SW6, CS18_SW6, CS16_SW6}, /* RGB92-sp */
+    {0, CS26_SW6, CS27_SW6, CS25_SW6}, /* RGB93-sp2 */
+    {0, CS29_SW6, CS30_SW6, CS28_SW6}, /* RGB94-ralt */
+    {0, CS32_SW6, CS33_SW6, CS31_SW6}, /* RGB95- fn */
+    {0, CS35_SW6, CS36_SW6, CS34_SW6}, /* RGB96-left */
+    {0, CS38_SW6, CS39_SW6, CS37_SW6}, /* RGB97-down */
+    {0, CS11_SW9, CS12_SW9, CS10_SW9}, /* RGB98-right */
+    {0, CS14_SW9, CS15_SW9, CS13_SW9}, /* RGB99- 0 */
+    {0, CS20_SW9, CS21_SW9, CS19_SW9}  /* RGB100- . */
+};
+
+bool process_record_kb(uint16_t keycode, keyrecord_t* record) {
+    if (record->event.pressed) {
+        switch (keycode) {
+            case QK_KB_0:
```

Comment:
For reference, here is how the normal keycodes function: 
https://github.com/qmk/qmk_firmware/blob/master/quantum/process_keycode/process_rgb.c

If you're fine with this, awesome. But just in case you want to have them handled like the normal RGB keycodes, you can do so.

Reply:
Yes, we do  want to have them handled like the normal RGB keycodes. Thank you very much for your suggestion anyway.